skbio/stats/power.py

```diff
@@ -295,9 +307,22 @@ def subsample_power(test, samples, draw_mode='ind', alpha_pwr=0.05,
                 raise ValueError('Each vector in samples must be the same '
                                  'length in "matched" draw_mode.')
 
-    # Checks there are enough samples to subsample
-    if num_ids <= min_observations:
-        raise ValueError('There are not enough samples for subsampling.')
+    # Checks the ratio
+    num_groups = len(samples)
+    if ratio is None:
+        ratio = np.ones((num_groups))
+    ratio = np.asarray(ratio)
+    if not ratio.shape == (num_groups,):
+        raise ValueError('There must be a ratio for each group.')
+
+    # Determines the number of p-values generated by the statistical test
+    p_return = test(samples)
+    if isinstance(p_return, float):
+        num_p = 1
+    elif isinstance(p_return, np.ndarray) and len(p_return.shape) == 1:
+        num_p = p_return.shape[0]
+    else:
+        raise TypeError('test must return a float or one-dimensional array.')
```

Comment:
Not a fan of returning different types depending on the input to `test`, but I don't have a suggestion right now (wanted to add a note to keep track of this :)